### PR TITLE
changes

### PR DESCRIPTION
**Pull request by jessemtso**
damien, look at these

### CODE REVIEW

File: dao.py [hunk 1]
```diff
@@ -267,13 +267,27 @@ def get_tournament_by_id(self, id):
...
             # remove everything before the last special character; hopefully removes crew/sponsor tags
-            re.split(special_chars, alias_lower)[-1].strip()
+            re.split(special_chars, alias_lower)[-1].strip(),
+            # regex nonsense to deal with pool prefixes
+            re.split(re_test_1, alias_lower)[2],
+            re.split(re_test_2, alias_lower)[2],
+            # well, we're using set, so why not
+            re.split(re_test_1, alias_lower)[2].strip(),
+            re.split(re_test_2, alias_lower)[2].strip(),
+            # because fuck it
```
Comment (jiangtyd) [inline dao.py]: eh kill this comment lol

File: dao.py [hunk 2]
```diff
@@ -267,13 +267,27 @@ def get_tournament_by_id(self, id):
...
+            re.split(" ", alias_lower)
```
Comment (jiangtyd) [inline dao.py]: isn't this a list? don't we want a `str`

File: server.py [hunk 3]
```diff
@@ -431,7 +431,12 @@ def get(self, region, id):
         if tournament is not None:
             response = convert_tournament_to_response(tournament, dao)
         else:
+            user = get_user_from_request(request, dao)
```
Comment (jiangtyd) [inline server.py]: this is a separate change, would prefer to put it in a separate PR

File: server.py [hunk 4]
```diff
@@ -480,28 +485,21 @@ def put(self, region, id):
...
-            if not tournament:
-                return "No pending tournament found with that id.", 400
-            else:
-                if not is_user_admin_for_regions(user, tournament.regions):
-                    return 'Permission denied', 403
-                else:
-                    return "Cannot delete a finalized tournament.", 400
-
-        if not is_user_admin_for_regions(user, pending_tournament.regions):
+        tournament_to_delete = dao.get_pending_tournament_by_id(ObjectId(id))
```
Comment (jiangtyd) [inline server.py]: also a separate change

File: server.py [hunk 5]
```diff
@@ -480,28 +485,21 @@ def put(self, region, id):
...
+        if not tournament_to_delete: #if there's no pending tournament with that ID, it must be a finalized tournament
+            tournament_to_delete = dao.get_tournament_by_id(ObjectId(id))
+        if not tournament_to_delete: #can't find anything, whoops
+            return "No tournament (pending or finalized) found with that id.", 400
+        if not is_user_admin_for_regions(user, tournament_to_delete.regions):
```
Comment (jiangtyd) [inline server.py]: maybe we should do the `403` check first

Comment (jiangtyd) [inline server.py]: wait nvm we have to get it first to get the region

**Comment by jiangtyd** [comment]
ok i'll give you the "multiple changes in one PR" thing for now I guess, but I don't think you should merge the `DELETE tournament` change without tests

**Comment by jessemtso** [comment]
closing this PR, will try and break things down into more fine grained pieces
